### PR TITLE
feat: LEAP-1110: Improve events and update docs for events

### PR DESCRIPTION
**Pull request by Gondragos**
- update docs about new added events
- fix doc about `selectAnnotation` event
- add useful information for `selectHistory` and `beforeSaveAnnotation` events

### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend

### Does this PR introduce a breaking change?
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [ ] No
- [x] Not sure (briefly explain the situation below)

It could but we didn't use arguments from `selectHistory` yet.

### Which logical domain(s) does this change affect?
`Docs`, `Events`

### CODE REVIEW

File: docs/source/guide/frontend_reference.md [hunk 1]
```diff
@@ -285,6 +285,18 @@ Draft is sent to the server.
...
+### `beforeSaveAnnotation`
+Annotation is going to be saved as the result of the `submit` or `update` action. 
+Returning `false` from this event will prevent saving the annotation.
+
+| Argument         | Type     | Description                                                                                                              |
+|------------------|----------|--------------------------------------------------------------------------------------------------------------------------|
+| `labelStudio`    | `Object` | Instance of Label Studio                                                                                                 |
+| `annotation`     | `Object` | Current annotation                                                                                                       |
+| `payload`        | `Object` | Additional information                                                                                                   |
+| `payload.event`  | `string` | Indicates which event is going to be next (`submitAnnotation`, `updateAnnotation`) |
```
Comment (hlomzik) [inline docs/source/guide/frontend_reference.md]:
```suggestion
|------------------|----------|---------------------------------------------------------------------------------------------|
| `labelStudio`    | `Object` | Instance of Label Studio                                                                                                 |
| `annotation`     | `Object` | Current annotation                                                                                                       |
| `payload`        | `Object` | Additional information                                                                                                   |
| `payload.event`  | `string` | Indicates which event is about to be executed (`submitAnnotation`, `updateAnnotation`, etc.) |
```

**Review by hlomzik — APPROVED** [review]

**Comment by Gondragos** [comment]
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/9388439156)
> Successfully merged:  10 files changed, 115 insertions(+), 15 deletions(-)

File: docs/source/guide/frontend_reference.md [hunk 2]
```diff
@@ -307,10 +319,13 @@ Annotation is updated.
...
 
-| Argument      | Type     | Description              |
-| ------------- | -------- | ------------------------ |
-| `labelStudio` | `Object` | Instance of Label Studio |
-| `annotation`  | `Object` | Current annotation       |
+| Argument              | Type      | Description                                   |
+|-----------------------|-----------|-----------------------------------------------|
+| `annotation`          | `Object`  | Current annotation                            |
+| `previousAnnotation`  | `Object`  | Previous annotation                           |
+| `payload`             | `Object?` | Additional information                        |
```
Comment (caitlinwheeless) [inline docs/source/guide/frontend_reference.md:326]: Check question marke - and could we just say "Payload content" or something under Description

Comment (caitlinwheeless) [inline docs/source/guide/frontend_reference.md:326]:
```suggestion
| `payload`             | `Object` | Payload content                        |
```

**Pull request merged**